assets/model_monitoring/components/model_monitor/model_monitor_metric_publisher/spec.yaml

```diff
@@ -0,0 +1,62 @@
+$schema: http://azureml/sdk-2-0/SparkComponent.json
+type: spark
+
+name: model_monitor_metric_outputter
+display_name: Model Monitor - Metric Publisher
+description: Azure Monitor Publisher for the computed model monitor metrics.
+version: 0.3.6
+is_deterministic: true
+
+code: ../../src/
+entry:
+  file: ./model_monitor_metric_publisher/run.py
+
+inputs:
+  signal_metrics:
+    type: mltable
+    mode: direct
+  monitor_name:
+    type: string
+  signal_name:
+    type: string
+  location:
+    type: string
+  workspace_resource_id:
```

Comment:
do you have that information from within the component?

Reply:
I did find the `workspace_resource_id` from the runtime env vars (`os.environ['AZUREML_WORKSPACE_SCOPE']`), assuming the workspace running the job is always going to be the same where the metrics go. Though I am unsure how reliable this is in future. To me, it is easier to construct (or control) in MoMo.

Reply:
After a second thought, both of the variables seem to be generally useful for any jobs. For simplicity of the caller, I changed to pull the location and workspace resource id from the env vars, with some assertion to make sure they exists and are valid.